aten/src/ATen/TensorIterator.cpp

```diff
@@ -808,18 +808,22 @@ void TensorIteratorBase::select_all_keeping_dim(int start_dim, IntArrayRef indic
   }
 }
 
+TensorIteratorConfig TensorIteratorBase::binary_op_config(const Tensor& out, const Tensor& a, const Tensor& b) {
+  TensorIteratorConfig config;
+  config.set_check_mem_overlap(true)
+        .add_output(out)
+        .add_input(a)
+        .add_input(b)
+        .allow_cpu_scalars(true)
+        .promote_inputs_to_common_dtype(true)
+        .cast_common_dtype_to_outputs(true)
+        .enforce_safe_casting_to_output(true);
+  return config;
+}
+
 TensorIterator TensorIterator::binary_op(Tensor& out, const Tensor& a,
     const Tensor& b) {
-  return TensorIteratorConfig()
-     .set_check_mem_overlap(true)
-     .add_output(out)
-     .add_input(a)
-     .add_input(b)
-     .allow_cpu_scalars(true)
-     .promote_inputs_to_common_dtype(true)
-     .cast_common_dtype_to_outputs(true)
-     .enforce_safe_casting_to_output(true)
-     .build();
+  return binary_op_config(out, a, b).build();
```

Comment:
why is this one factored and not the others?

Reply:
Because I only did add. Would make sense to do the rest of them in one go in a separate PR.